index.bs

```diff
@@ -8215,85 +8370,102 @@ for the specific requirements that the use of
-must either
+The [{{Exposed}}] [=extended attribute=] must either
 [=takes an identifier|take an identifier=] or
 [=takes an identifier list|take an identifier list=].
-Each of the identifiers mentioned must be
-a [=global name=].
+Each of the identifiers mentioned must be a [=global name=] and be unique.
+This list of identifiers is known as the construct's
+<dfn>own exposure set</dfn>.
 
-Every construct that the [{{Exposed}}]
-[=extended attribute=]
-can be specified on has an <dfn id="dfn-exposure-set" export>exposure set</dfn>,
-which is a set of [=interfaces=]
-defining which global environments the construct can be used in.
-The [=exposure set=]
-for a given construct is defined as follows:
+<div algorithm>
 
-*   If the [{{Exposed}}]
-    [=extended attribute=]
-    is specified on the construct, then the [=exposure set=]
-    is the set of all interfaces that have a [=global name=]
-    that is listed in the extended attribute's argument.
-*   If the [{{Exposed}}]
-    [=extended attribute=]
-    does not appear on a construct, then its
-    [=exposure set=] is defined
-    implicitly, depending on the type of construct:
+    To get the <dfn id="dfn-exposure-set" export>exposure set</dfn> of a construct |C|,
+    run the following steps:
+
+    1.  If |C| is not an [=interface=], [=namespace=], [=interface member=], [=mixin member=],
+        or [=namespace member=], then return null.
+    1.  Let |H| be |C|'s [=host interface=] if |C| is a [=mixin member=], or null otherwise.
+    1.  If |C| is an [=interface member=], [=mixin member=], or [=namespace member=], then:
+        1.  If the [{{Exposed}}] [=extended attribute=] is specified on |C|, then:
+            1.  If |H| is set, return the [=set/intersection=] of |C|'s [=own exposure set=]
+                and |H|'s [=exposure set=].
+            1.  Otherwise, return |C|'s [=own exposure set=].
+        1.  Otherwise, set |C| to be the
+            [=interface=], [=partial interface=],
+            [=mixin=], [=partial mixin=],
+            [=namespace=], or [=partial namespace=]
+            |C| is declared on.
+    1.  If |C| is a [=partial interface=], [=partial mixin=], or [=partial namespace=], then:
+        1.  If the [{{Exposed}}] [=extended attribute=] is specified on |C|, then:
+            1.  If |H| is set, return the [=set/intersection=] of |C|'s [=own exposure set=]
+                and |H|'s [=exposure set=].
+            1.  Otherwise, return |C|'s [=own exposure set=].
+        1.  Otherwise, set |C| to be the original [=interface=], [=mixin=], or [=namespace=]
+            definition of |C|.
+    1.  If |C| is a [=mixin=], then:
+        1.  If the [{{Exposed}}] [=extended attribute=] is specified on |C|, then:
+            1.  Return the [=set/intersection=] of |C|'s [=own exposure set=]
+                and |H|'s [=exposure set=].
+        1.  Otherwise, return |H|'s [=exposure set=].
+    1.  If |C| is a [=interface=] or [=namespace=], then:
```

Comment:
Nit: Un-indent the following lines and convert this into an "Assert: |C| is either an interface or a namespace"